Utils/azuretools-core/src/com/microsoft/azuretools/utils/NetUtils.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azuretools.utils;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.Enumeration;
+
+/**
+ * Networking-related convenience methods.
+ */
+public final class NetUtils {
```

Comment:
May use `AppInsightsClient.getInstallationId()` to get the installation id, which is the hash of mac address

Reply:
good.